### ISSUE TITLE
MSAA: overriding CreateAccessibilityInstance() with custom AccessibleObject implementation does not work and causes InvalidOperationException 

### ISSUE THREAD

**Issue by tbambuch** [dotnet/winforms#11831]
### .NET version

8.0

### Did it work in .NET Framework?

Yes

### Issue description

If I understand the docs correctly (and it used to work in the Framework, so I believe I do), the correct way to implement custom MSAA / UI Automation provider is to override CreateAccessibilityInstance() and pass the AccessibleObject.
However when I return an AccessibleObject which is just derived from ControlAccessibleObject without overriding anything, the MSAA is not reported correctly and uses the default values for generic Control. Overriding anything like Role or Status properties has no effect on MSAA at all, but is correctly reported in UIA.
Also, when overriding CreateAccessibilityInstance(), the InvalidOperationException is raised sometimes.

### Steps to reproduce

Sample is attached

- Create a custom control. In my case I derived MyCheckBox from CheckBox.
- Set Accessible role to anything than default CheckBox´s role - in my case `AccessibleRole = AccessibleRole.Slider`
- If you run the app now and inspect the accessiblity with Inspect.exe, everything works as expected both for MSAA and UIA
- Now override CreateAccessibilityInstance() and return a new ControlCheckBoxAccessibleObject() which just derives from Winforms CheckBoxAccessibleObject, without overriding anything.
- If you run the app now, MSAA reports the role as "Window", however UIA reports the correct type - "Slider"

Also, if you click on the checkbox, InvalidOperationException is raised:

`Type 'AccessibilityTest.ControlCheckBoxAccessibleObject' has a ComVisible(false) parent 'System.Windows.Forms.CheckBox+CheckBoxAccessibleObject' in its hierarchy, therefore QueryInterface calls for IDispatch or class interfaces are disallowed.`'

[AccessibilityTest.zip](https://github.com/user-attachments/files/16520429/AccessibilityTest.zip)

**Comment by JeremyKuhne** [comment]
@lonitra this might be moot with the switch to ComWrappers?

**Comment by lonitra** [comment]
Yep, tested this with SDK 9.0.100-rc.1.24405.3 and this no longer repros.

**Comment by JeremyKuhne** [comment]
@tbambuch we rewrote our COM code to use ComWrappers in .NET 9, which has fixed this issue.
